tests/ssl-opt.sh

```diff
@@ -4863,8 +4867,176 @@ run_test    "Record Size Limit: TLS 1.3: Client-side parsing, debug output and f
 #            -s "Sending extension Record Size Limit/28 (2 bytes)" \
 #            -c "EncryptedExtensions: record_size_limit(28) extension received."\
 #            -c "found record_size_limit extension" \
-#            -c "RecordSizeLimit: 16385 Bytes" \
-#            -s "Received alert \[110]: An unsupported extension was sent"
+#            -c "RecordSizeLimit: 16385 Bytes"
+
+# In the following (9) tests, --recordsize is the value used by the G_NEXT_CLI (3.7.2) to configure the
+# maximum record size using "https://gnutls.org/reference/gnutls-gnutls.html#gnutls-record-set-max-size".
+# There is currently a lower limit of 512, caused by this function not respecting the
+# "%ALLOW_SMALL_RECORDS" priority string and not using the more recent function
+# https://gnutls.org/reference/gnutls-gnutls.html#gnutls-record-set-max-recv-size.
+# There is currently an upper limit of 4096, caused by the cli arg parser:
+# https://gitlab.com/gnutls/gnutls/-/blob/3.7.2/src/cli-args.def#L395.
+# Thus, these tests are currently limit to that value range.
+# Moreover, the value sent in the extension is expected to be larger by one compared
+# to the value passed on the cli:
+# https://gitlab.com/gnutls/gnutls/-/blob/3.7.2/lib/ext/record_size_limit.c#L142
+
+# Currently test certificates being used do not fit in 513 record size limit
+# so 513 record size limit tests will not pass until certificates size
+# is reduced or handshake messages fragmentation is supported.
+
+# requires_gnutls_tls1_3
+# requires_gnutls_record_size_limit
+# requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT
+# requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+# run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit (513), 1 fragment" \
+#             "$P_SRV debug_level=3 force_version=tls13 response_size=256" \
```

Comment:
I thought the tests with a record size limit of 512 should fail. But looking at the CI (and locally for me) it does not seem the case. What am I missing?

Reply:
I just missed that they are commented out ...

Reply:
The tests in this tocket will never fail becuase GnuTLS does not start to commit to the record size restrictions except when the server sends it's record size as well which GnuTLS treats as acknowledgment from the server to accepting the record size requested ( see https://gitlab.com/gnutls/gnutls/-/blob/master/lib/ext/record_size_limit.c?ref_type=heads#L90 ) so tests will start to fail in https://github.com/Mbed-TLS/mbedtls/pull/8526

Reply:
Ah that makes sense. Thanks.